docs/DEPLOYMENT.md

```diff
@@ -1,32 +1,116 @@
+6. AWS Virtual Private Cloud (VPC)
+7. AWS Fargate
+8. AWS Elastic Load Balancer (ELB)
+9. AWS Elastic IPs
+10. AWS Identity and Access Management (IAM)
+11. AWS Relational Database Service (RDS)
 
 #### Staging
-```sh
-# RUN THIS COMMAND ONLY IF STATIC FILES HAVE CHANGED
-zappa manage dev "collectstatic --noinput"
-zappa update dev
-```
+Setting up staging environment on AWS is pretty straightforward.
+1. Login to your AWS console.
+2. Go to VPC. (skip this step if you've already created a VPC when setting up frontend repository)
+   1. Create a new VPC.
+   2. Name it `plio-staging-vpc`.
+   3. In IPv4 CIDR block, enter `10.0.0.0/28`.
+   4. Click on create button. You will see the new VPC under the list of VPCs.
+   5. Check out this [AWS guide for more details on VPC](https://docs.aws.amazon.com/vpc/latest/userguide/working-with-vpcs.html).
+3. Set up the database. Click on `Databases` on the AWS RDS page.
+   1. Click on create database.
+   2. Use "Standard create" in database creation method.
+   3. Use Postgres12 in the DB engine.
+   4. Under templates, go with "Production".
+   5. Name the DB instance identifier as `plio`.
+   6. Set the master user name and password.
+   7. For DB instance, we can go with Burstable classes to get started with. Select the one having vCPUs & RAM as per your needs.
+   8. Set the alloted storage as per your needs.
+   9. For Availability and Durability, use multi-AZ deployment if you need. Take a note that this may almost double your monthly expense.
+   10. Under VPC settings, select the VPC created in previous step.
+   11. In case you prefer to connect to your database directly from your machine, set the public access to "Yes".
+   12. Under additional configuration, set initial database name as `plio_staging`. This will create an empty database.
+   13. Finally, create the RDS instance by clicking on "Create Database" button.
+   14. You will see the new RDS instance in the list of RDS instances. Click on your DB instance to see the connectivity settings.
+   15. To connect to RDS from command line, run the following command (this will work if you've set public access to yes):
+   ```sh
+   psql -h your-db-instance-endpoint.aws-region.rds.amazonaws.com -p 5432 -U master_username
+   ```
+   16. Once you are logged in into the PSQL CLI, you can run all the SQL commands to create a new database, user, grant privileges etc.
+4. Create a new Elastic IP by going to EC2 dashboard and navigating to Elastic IP section.
+   1. Click on Allocate Elastic IP address and click on Allocate button.
+   2. You will see a new IP address in the IPs list. Name it `plio-backend-staging`.
+   3. This will be used in later steps to give the load balancer a permanent IP address.
+5. Go to Target groups.
+   1. Create a new target group.
+   2. Choose target type to be `IP addresses`.
+   3. Name the target group as `plio-backend-staging`.
+   4. Select the `plio-staging-vpc` for the target group VPC.
```

Comment:
I had to add an IP address in this step as a target

Reply:
Protocol needs to be changed from HTTP to TCP

Reply:
done with protocol update.

I'm not sure about the IP. I hadn't set it when creating the target group for staging and I don't think it's a required field, right?